pkg/util/proxyutil/proxyutil.go

```diff
@@ -7,9 +7,17 @@ import (
 )
 
 // PrepareProxyRequest prepares a request for being proxied.
-// Removes X-Forwarded-Host, X-Forwarded-Port, X-Forwarded-Proto headers.
+// Removes X-Forwarded-Host, X-Forwarded-Port, X-Forwarded-Proto, Origin, Referer headers.
+// Set X-Grafana-Referer based on contents of Referer.
 // Set X-Forwarded-For headers.
 func PrepareProxyRequest(req *http.Request) {
+	// Set X-Grafana-Referer to correlate access logs to dashboards
+	req.Header.Set("X-Grafana-Referer", req.Header.Get("Referer"))
```

Comment:
As we are re-using as is the Referer value that is an untrusted user input, it could be useful in the unit tests to make sure something like
`Referer: https://www.google.ch\nOtherHeader:https://www.somethingelse.com` does not get processed as 2 different headers: `Referer` and `OtherHeader`

Reply:
A test with this would work, but I'm guessing that wouldn't be checking what you're thinking of `req.Header.Set("Referer", "https://www.google.ch\r\nOtherHeader:https://www.somethingelse.com")`.

I could also add something like this, but feels like we'd be testing request parsing more than Grafana.  But then it's effectively another header anyway. 
```
t.Run("Prepare proxy request should set X-Grafana-Referer header as one header", func(t *testing.T) {
  str := "GET / HTTP/1.1\r\n" +
	"Host: blah.com\r\n" +
	"Accept: text/html\r\n" +
	"Referer: https://www.google.ch\r\nOtherHeader:https://www.somethingelse.com\r\n" +
	"Accept-Encoding: gzip, deflate\r\n" +
	"Connection: close\r\n" +
	"\r\n" +
	""
  buf := bufio.NewReader(strings.NewReader(str))
  
  req, err := http.ReadRequest(buf)
  require.NoError(t, err)
  
  PrepareProxyRequest(req)
  require.Contains(t, req.Header, "X-Grafana-Referer")
  require.Equal(t, "https://www.google.ch", req.Header.Get("X-Grafana-Referer"))
})
```

Reply:
Testing  that `req.Header.Set("Referer", "https://www.google.ch\r\nOtherHeader:https://www.somethingelse.com")` is not producing 2 headers is enough: it "proves" we don't have to care about multiline in this untrusted input and hence we are protected from header injection

Reply:
Gotcha.  I added a test to check that in [95d01c6](https://github.com/grafana/grafana/pull/60040/commits/95d01c680e59969e0b2e808b35bc3f056d777fe3)

Reply:
Thank you. Could we add a `require.NotContains(t, req.Header, "OtherHeader")` to double check ?

Reply:
Sure thing, added in [b0a6df9](https://github.com/grafana/grafana/pull/60040/commits/b0a6df9a0a83d51ad55b113e0e8eb028a2cef792)

Reply:
I don't think so.  req.Header.Set is essentially just setting a value in a `map[string][]string`.  There's no special encoding/decoding there.   Maybe I'm misunderstanding the risk though.

Reply:
Okay just wanted a sanity check